packages/timeline-state-resolver/src/integrations/tricaster/triCasterTimelineStateConverter.ts

```diff
@@ -0,0 +1,155 @@
+import { TimelineState } from 'superfly-timeline'
+import {
+	Mappings,
+	MappingTriCaster,
+	MappingTriCasterType,
+	TSRTimelineObjBase,
+	isTimelineObjTriCasterAudioChannel,
+	isTimelineObjTriCasterDSK,
+	isTimelineObjTriCasterME,
+	isTimelineObjTriCasterMixOutput,
+	MappingTriCasterMixEffect,
+	MappingTriCasterDownStreamKeyer,
+	MappingTriCasterAudioChannel,
+	MappingTriCasterMixOutput,
+	TriCasterAudioChannelName,
+	TriCasterMixEffectName,
+	TriCasterMixOutputName,
+	MappingTriCasterInput,
+	isTimelineObjTriCasterInput,
+	TriCasterInputName,
+} from 'timeline-state-resolver-types'
+import * as _ from 'underscore'
+import { TriCasterState } from './triCasterStateDiffer'
+
+type DeepPartial<T> = { [P in keyof T]?: T[P] extends object ? DeepPartial<T[P]> : T[P] }
+
+export class TriCasterTimelineStateConverter {
+	private meNames: Set<TriCasterMixEffectName>
+	private inputNames: Set<TriCasterInputName>
+	private audioChannelNames: Set<TriCasterAudioChannelName>
+	private mixOutputNames: Set<TriCasterMixOutputName>
+
+	constructor(
+		private readonly getDefaultState: () => TriCasterState,
+		meNames: TriCasterMixEffectName[],
+		inputNames: TriCasterInputName[],
+		audioChannelNames: TriCasterAudioChannelName[],
+		mixOutputNames: TriCasterMixOutputName[]
+	) {
+		this.meNames = new Set(meNames)
+		this.inputNames = new Set(inputNames)
+		this.audioChannelNames = new Set(audioChannelNames)
+		this.mixOutputNames = new Set(mixOutputNames)
+	}
+
+	getTriCasterStateFromTimelineState(
+		timelineState: TimelineState,
+		newMappings: Mappings,
+		deviceId: string
+	): TriCasterState {
+		const resultState = this.getDefaultState()
+		const sortedLayers = this.sortLayers(timelineState)
+
+		_.each(sortedLayers, ({ tlObject, layerName }) => {
+			const mapping = newMappings[layerName] as MappingTriCaster | undefined
+			if (!mapping || mapping.deviceId !== deviceId) {
+				return
+			}
+			switch (mapping.mappingType) {
+				case MappingTriCasterType.ME:
+					this.applyMixEffectState(resultState, tlObject, mapping)
+					break
+				case MappingTriCasterType.DSK:
+					this.applyDskState(resultState, tlObject, mapping)
+					break
+				case MappingTriCasterType.INPUT:
+					this.applyInputState(resultState, tlObject, mapping)
+					break
+				case MappingTriCasterType.AUDIO_CHANNEL:
+					this.applyAudioChannelState(resultState, tlObject, mapping)
+					break
+				case MappingTriCasterType.MIX_OUTPUT:
+					this.applyMixOutputState(resultState, tlObject, mapping)
+					break
+			}
```

Comment:
```suggestion
			const applyState: (
				resultState: TriCasterState,
				tlObject: TSRTimelineObjBase,
				mapping: MappingTriCasterMixEffect) => void
			switch (mapping.mappingType) {
				case MappingTriCasterType.ME:
					applyState = this.applyMixEffectState
					break
				case MappingTriCasterType.DSK:
					applyState = this.applyDskState
					break
				case MappingTriCasterType.INPUT:
					applyState = this.applyInputState
					break
				case MappingTriCasterType.AUDIO_CHANNEL:
					applyState = this.applyAudioChannelState
					break
				case MappingTriCasterType.MIX_OUTPUT:
					applyState = this.applyMixOutputState
					break
			}
			applyState(resultState, tlObject, mapping)
```

The call under each case, could be extracted as shown in the suggestion, such that the case only choses which method to use. 

Had this been in newer versions of C#, the footprint of the code would shrink doing something like this. In this case it does grow slightly. 
Haven't tested if the suggested change is syntactically correct, but i do think it gets the idea accross.

Would reduce the duplication slightly, as '`(resultState, tlObject, mapping)`' would not have to be added multiple time.
Should also make it easier if a similar new case is added in the future.

Reply:
This is overengineering and now you are forcing all methods being called within each case to have the same arguments making it less flexible. Please don't do this.